### ISSUE TITLE
How to authenticate against Azure AD?

### ISSUE THREAD

**Issue by shizzor** [websanova/vue-auth#271]
I managed to install this library, this is well documented, but I am at a loss for the next steps.
How do I actually use it to authenticate users?
I saw somethings about some oauth providers, but there is not explanation or hint in how to actually use them or how to add more. How to I add a new "driver" or "strategy"?

How would I go about authenticating against something like Azure AD? Do I need to use something like passport or,for Azure, adal?

**Comment by websanova** [comment]
All the library does I'd send and receive tokens. It's nothing magical.
Thanks drivers just parse whatever comes in and format whatever goes out.

You would need to get some requirements first. Like url end points,
headers, parameters, etc for whatever your auth server is

**Comment by shizzor** [comment]
Yeah, I have all the requirements like endpoint, clientId and clientSecret. I've registered my app in their portal and I already have all of this working in a nodejs-only app.

Doing it with vue-auth is still rather confusing.

By now I managed to get a JWT token using ADAL.js and set the token and user in vue-auth using `this.$auth.token(null, token)` and `this.$auth.user(myUser)`. But then whenever I proceed to request a resource (from my API), vue-auth redirects to refresh the token and is losing it subsequently.

**Comment by websanova** [comment]
Well hard to help if you don't give some details ;)

Did you look at any of the existing auth drivers?

https://github.com/websanova/vue-auth/tree/master/drivers/auth

**Comment by shizzor** [comment]
Yes, I looked at them, and I figured they should actually suffice for simply adding the auth headers with the token to the request. I also saw implementations for google and facebook oauth and at first thought I would need to add something similar for azure.

Also, what I was wondering about was, when does your plugin decide to refresh tokens? I was not able to just hand the token over to vue-auth and let it add the appropriate headers. It would always try to refresh the token right away and lose the one I set.

Eventually I wrote my own plugin for authenticating against the Azure AD, since I had to do so much stuff  outside of what vue-auth provided.
